src/Clients/MainApp/FSpot/MainWindow.cs

```diff
@@ -1774,7 +1774,9 @@ void HandleAdjustTime (object sender, EventArgs args)
 			// this resolves "Invalid object type `GnomeDateEdit'" on loading the AdjustTimeDialog
 			var type = Gnome.DateEdit.GType;
 
-			(new AdjustTimeDialog (Database, list)).Run ();
+			var dialog = new AdjustTimeDialog (Database, list);
+			dialog.Run ();
+			dialog.Destroy ();
```

Comment:
Sounds like a good idea but does not work. With a `using` statement the dialog still requires a second click on the X button to be closed.